src/mpi/init/finalize.c

```diff
@@ -56,8 +56,8 @@ void MPIR_Add_finalize(int (*f) (void *), void *extra_data, int priority)
          * MPIR_Process.mpich_state to decide how to signal the error */
         (void) MPL_internal_error_printf("overflow in finalize stack! "
                                          "Is MAX_FINALIZE_FUNC too small?\n");
-        if (OPA_load_int(&MPIR_Process.mpich_state) == MPICH_MPI_STATE__IN_INIT ||
-            OPA_load_int(&MPIR_Process.mpich_state) == MPICH_MPI_STATE__POST_INIT) {
+        if (OPA_load_int(&MPIR_Process.mpich_state) > MPICH_MPI_STATE__PRE_INIT &&
+            OPA_load_int(&MPIR_Process.mpich_state) < MPICH_MPI_STATE__POST_FINALIZED) {
```

Comment:
`state >= MPICH_MPI_STATE__IN_INIT_1 && state <= MPICH_MPI_STATE__POST_INIT` looks easier to understand for me.

Reply:
My reasoning is to minimize the exposure of `MPICH_MPI_STATE__IN_INIT_x` since they are init-only internal stages.

Reply:
Then, `state > MPICH_MPI_STATE__PRE_INIT && state <= MPICH_MPI_STATE__POST_INIT` should be better. At least I don't know what `state < MPICH_MPI_STATE__POST_FINALIZED` means.

Reply:
If you don't strongly oppose, let's change it to `<= MPICH_MPI_STATE__POST_INIT`. `state < MPICH_MPI_STATE__POST_FINALIZED` implies existence of `MPICH_MPI_STATE__PRE_FINALIZED`, which does not exist.